File: src/test/programmers/python/level2/FindPrimeNumberInKthNumber.py
def convert(n, k):
    converted = ''
    while n > 0:
        n, mod = divmod(n, k)
        converted += str(mod)
    return converted[::-1]

def is_prime(n):
    if n == 1:
        return False

    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True

def solution(n, k):
    prime_number = convert(n, k)
    nums = prime_number.split("0")
    answer = 0
    for i in nums:
        if i.isdigit():
            if is_prime(int(i)):
                answer += 1

    return answer

solution(437674, 3)

File: src/test/programmers/java/level2/FindPrimeNumberInKthNumber.java
package test.programmers.java.level2;

public class FindPrimeNumberInKthNumber {
    public int solution(int n, int k) {
        String[] nums = convert(n, k).split("0");
        int answer = 0;

        for (String num : nums) {
            if (isDigit(num)) {
                if (isPrime(Long.parseLong(num))) {
                    answer++;
                }
            }
        }

        return answer;
    }

    static boolean isDigit(String n) {
        return n.matches("[0-9]+");
    }

    static boolean isPrime(long n) {
        if (n == 1) {
            return false;
        }

        int range = ((int)Math.sqrt(n)) + 1;
        for (int i = 2; i < range; i++) {
            if (n % i == 0) {
                return false;
            }
        }

        return true;
    }

    static String convert(int n, int k) {
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            int mod = n % k;
            sb.append(mod);
            n /= k;
        }

        return sb.reverse().toString();
    }
}
